internal/support/logging/server.go

```diff
@@ -149,16 +164,16 @@ func getLogs(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	logs, err := dbClient.find(*criteria)
+	logs, err := dbClient.FindLog(*criteria, criteria.Limit)
 	if err != nil {
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
-	if criteria.Limit > 0 && len(logs) > criteria.Limit {
+	if criteria.Limit > 0 && len(logs) > Configuration.Service.ReadMaxLimit {
```

Comment:
You are ignoring the limit from the query and only testing if the log count is lower than the configured ReadMaxLimit. 

You should compare the limit read in sever.go:80 and if it is greater than ReadMaxLimit set it to ReadMaxLimit in the getCriteria function.

Reply:
OK, I can do that. But then why do we ever need to prepare the StatusRequestEntityTooLarge response? It never will be too large with your approach.

Reply:
I'm sorry, I didn't realize that you are limiting the size of the results in the findLog. It is ok as is.